.github/workflows/test.yml

```diff
@@ -289,5 +293,345 @@ jobs:
       - name: Show test directory
         run: bash -c "ls -ld $GITHUB_WORKSPACE/tests/*"
 
+      # nested LXD configuration
+      - name: Init LXD inside container
+        run: sudo lxc exec test-container -- sudo lxd init --preseed < $GITHUB_WORKSPACE/.github/resources/lxd4-init.yml
```

Comment:
How do you know it's version 4?

Reply:
Probably that's the version installed from the Github actions container.

Reply:
Do you have the answer @AnvithLobo

Reply:
oh I wrote it as a dirty test. You can use the same LXD installation logic the original test uses. I didn't bother as anything above 18.04 ubuntu will install 4.0 by default 

https://github.com/andreasscherbaum/ansible-lxc-ssh/blob/master/.github/workflows/test.yml#L74-#L80

Reply:
The "original" test makes sure to check the version: https://github.com/andreasscherbaum/ansible-lxc-ssh/blob/master/.github/workflows/test.yml#L80

It also fails (for now) if a newer version is found: https://github.com/andreasscherbaum/ansible-lxc-ssh/blob/master/.github/workflows/test.yml#L72
The GH Actions tests run on "ubuntu-latest" which will automatically pull in newer releases, which might be version 4, or might be a higher version. My original design concept for the tests was to make sure I only test known versions and fail early if newer versions are found which need more testing.

Reply:
yeah it's a good idea to have version checking for tests. 
So we would need a lot of duplicate code for the host and the first lxc container. 
I'm not currently sure if there is any type of functions / ansible roles of sorts for github actions that would let us reuse the code.
if not might need to just write a bunch of duplicate tasks.

Reply:
I'm also not aware of such roles.